libfwupdplugin/fu-self-test.c

```diff
@@ -3201,6 +3202,72 @@ fu_bios_settings_load_func(void)
 		g_debug("%s", tmp);
 		g_assert_null(g_strrstr(tmp, "[Status"));
 	}
+#else
+	g_assert_error(error, G_FILE_ERROR, G_FILE_ERROR_NOENT);
+	g_assert_false(ret);
+	g_clear_error(&error);
+#endif
+	g_free(test_dir);
+
+	/* load BIOS settings from a Lenovo P620 running 6.3 */
+	test_dir =
+	    g_test_build_filename(G_TEST_DIST, "tests", "bios-attrs", "lenovo-p620-6.3", NULL);
+	(void)g_setenv("FWUPD_SYSFSFWATTRIBDIR", test_dir, TRUE);
+
+	ret = fu_context_reload_bios_settings(ctx, &error);
+	g_assert_no_error(error);
+	g_assert_true(ret);
+
+	p620_6_3_settings = fu_context_get_bios_settings(ctx);
+	p620_6_3_items = fu_bios_settings_get_all(p620_6_3_settings);
+	g_assert_cmpint(p620_6_3_items->len, ==, 129);
+
+	/* make sure nothing pending */
+	ret = fu_context_get_bios_setting_pending_reboot(ctx);
+	g_assert_false(ret);
+
+	/* check a BIOS setting reads from kernel 6.3 as expected by fwupd */
+	setting = fu_context_get_bios_setting(ctx, "com.thinklmi.AMDMemoryGuard");
+	g_assert_nonnull(setting);
+	tmp = fwupd_bios_setting_get_name(setting);
+	g_assert_cmpstr(tmp, ==, "AMDMemoryGuard");
+	tmp = fwupd_bios_setting_get_description(setting);
+	g_assert_cmpstr(tmp, ==, "AMDMemoryGuard");
+	tmp = fwupd_bios_setting_get_current_value(setting);
+	g_assert_cmpstr(tmp, ==, "Disable");
+	values = fwupd_bios_setting_get_possible_values(setting);
+	for (guint i = 0; i < values->len; i++) {
+		const gchar *possible = g_ptr_array_index(values, i);
+		if (i == 0)
+			g_assert_cmpstr(possible, ==, "Disable");
+		if (i == 1)
+			g_assert_cmpstr(possible, ==, "Enable");
+	}
+
+	/* try to read an BIOS setting known to have ][Status] to make sure we worked
+	 * around the thinklmi bug sufficiently
+	 */
+	setting = fu_context_get_bios_setting(ctx, "com.thinklmi.StartupSequence");
+	g_assert_nonnull(setting);
+	tmp = fwupd_bios_setting_get_current_value(setting);
+	g_assert_cmpstr(tmp, ==, "Primary");
+	values = fwupd_bios_setting_get_possible_values(setting);
+	for (guint i = 0; i < values->len; i++) {
+		const gchar *possible = g_ptr_array_index(values, i);
+		if (i == 0)
+			g_assert_cmpstr(possible, ==, "Primary");
+		if (i == 1)
+			g_assert_cmpstr(possible, ==, "Automatic");
+	}
+
+	/* check no BIOS settings have [Status in them */
+	for (guint i = 0; i < p620_6_3_items->len; i++) {
+		setting = g_ptr_array_index(p620_6_3_items, i);
+		tmp = fwupd_bios_setting_get_current_value(setting);
+		g_debug("%s", tmp);
+		// Waiting for kernel patch
+		//		g_assert_null(g_strrstr(tmp, "[Status"));
```

Comment:
??

Reply:
/sigh 

There's one more place the unit tests are still failing it seems and I missed it, thanks for catching it.
These 3 attributes have a `ShowOnly` appended.

```
thinklmi/attributes/AlarmTime/current_value:[00:00:00][Status:ShowOnly]
thinklmi/attributes/AlarmDate/current_value:[01/01/2019][Status:ShowOnly]
thinklmi/attributes/UserDefinedAlarmTime/current_value:[00:00:00][Status:ShowOnly]
```

@mrhpearson the kernel should be stripping that too.

Reply:
You're right I guess since they're read only it probably doesn't really matter too much.  Let me adjust the test suite accordingly.